web/src/browser_history.js

```diff
@@ -13,7 +14,7 @@ export const state = {
     // hashes are changed without calling `hashchanged` in many ways.
     spectator_old_hash: hash_util.is_spectator_compatible(window.location.hash)
         ? window.location.hash
-        : "#",
+        : `#${user_settings.default_view}`,
```

Comment:
Why is this the default view hash rather than the empty string?

Reply:
We want to completely control who can set empty hash to avoid its presence in the URL.

`return_to_web_public_hash` uses `window.location.hash` since we want to trigger hashchange in that case.

Reply:
Which means this cannot be an empty hash or just "", otherwise an empty hash will be set.

Reply:
We’re at top level here. `user_settings.initialize_user_settings` hasn’t been called yet, so `user_settings.default_view` is `undefined`. ([Discussion](https://chat.zulip.org/#narrow/stream/6-frontend/topic/browser_history.20uses.20user_settings.20too.20early/near/1637379).)